src/lib/ui/core/Share/Share.svelte

```diff
@@ -0,0 +1,44 @@
+<script lang="ts">
+  import Button from '$ui/core/Button/index.js'
+  import { cn } from '$ui/utils/index.js'
+  import { copy } from '$lib/utils/clipboard.js'
+
+  import { showShareDialog$ } from './ShareDialog.svelte'
+
+  type TProps = {
+    class?: string
+    getLink: () => Promise<string>
+  }
+
+  const { class: className = '', getLink }: TProps = $props()
+
+  let explanation = $state('Copy link')
+
+  const showShareDialog = showShareDialog$()
+
+  function onShareClick() {
+    getLink().then((link) => {
+      showShareDialog({ data: { link }, feature: 'chart_layout', source: 'charts' })
+    })
+  }
+
+  function onCopyLinkClick() {
```

Comment:
Better to add generic analytics here: `trackEvent('press', { type: "quick_link_copy", source })`